### PR TITLE
[15.0][MIG] project_timeline: Migration to 15.0

### PR DESCRIPTION
**Pull request by cesar-tecnativa**
@Tecnativa TT37717

### CODE REVIEW

**Comment by cesar-tecnativa** [comment]
Deppends on https://github.com/OCA/web/pull/2247

File: project_timeline/views/project_task_view.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                colors="white: user_ids == []; #2ecb71: kanban_state == 'done'; #ec7063: kanban_state == 'blocked'"
+            >
+                <field name="user_ids" />
+                <field name="planned_hours" />
+                <templates>
+                    <t t-name="timeline-item">
+                        <div class="o_project_timeline_item">
+                            <img
+                                t-if="record.user_ids"
+                                t-attf-src="/web/image/res.users/#{record.user_ids[0][0]}/image_128/16x16"
```
Comment (chrisandrewmann) [inline project_timeline/views/project_task_view.xml]: I had to change to user_ids[0] to give ID or no image displayed:

`t-attf-src="/web/image/res.users/#{record.user_ids[0]}/image_128/16x16"`

File: project_timeline/views/project_task_view.xml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                                t-att-title="record.user_ids[0][1]"
```
Comment (chrisandrewmann) [inline project_timeline/views/project_task_view.xml]: Removed t-att-title as not working (user_ids is only a list of IDs apparently)

**Comment by chrisandrewmann** [comment]
Also noticed after doing some testing that timeline crashes when grouping by a many2many field (Assignees) for example, with error below.

UncaughtClientError > TypeError
Uncaught Javascript Error > Cannot read properties of undefined (reading 'localeCompare')
TypeError: Cannot read properties of undefined (reading 'localeCompare')
    at group_order (http://localhost:8069/web/assets/681-7ac6ca2/web.assets_backend.min.js:10370:21)
    at Array.sort (<anonymous>)
    at DataSet._sort (http://localhost:8069/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:14783:24)
    at DataSet.getIds (http://localhost:8069/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:14642:17)
    at ItemSet._orderGroups (http://localhost:8069/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:32446:41)
    at ItemSet.redraw (http://localhost:8069/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:31842:23)
    at http://localhost:8069/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:24820:30
    at Array.forEach (<anonymous>)
    at Timeline._redraw (http://localhost:8069/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:24819:48)
    at Timeline._redraw (http://localhost:8069/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:37574:31)

File: project_timeline/views/project_task_view.xml (hunk 1 above)
Comment (jguenat) [reply project_timeline/views/project_task_view.xml]: You can display all user_ids images with this:

`<t t-foreach="record.user_ids" t-as="user_id">
    User
</t>`

File: project_timeline/views/project_task_view.xml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        <field name="model">project.task</field>
+        <field name="type">timeline</field>
+        <field name="arch" type="xml">
+            <timeline
+                date_start="date_start"
+                date_stop="date_end"
+                default_group_by="project_id"
+                event_open_popup="true"
+                colors="white: user_ids == []; #2ecb71: kanban_state == 'done'; #ec7063: kanban_state == 'blocked'"
+            >
```
Comment (chrisandrewmann) [inline project_timeline/views/project_task_view.xml:20]: Seeing as V15 now has task dependency built-in, could I suggest we add the following to put the arrows on tasks that use this feature:

```
                colors="white: user_ids == []; #2ecb71: kanban_state == 'done'; #ec7063: kanban_state == 'blocked'"
                dependency_arrow="depend_on_ids"
            >
```

**Comment by pedrobaeza** [comment]
Please rebase and push for forcing the rebuild.

**Comment by CarlosRoca13** [comment]
Done @pedrobaeza

File: project_timeline/views/project_task_view.xml (hunk 3 above)
Comment (CarlosRoca13) [reply project_timeline/views/project_task_view.xml:20]: Adding what you say, the timeline will look like this:
imagen

I think the timeline is a bit overloaded

Comment (chrisandrewmann) [reply project_timeline/views/project_task_view.xml:20]: Yeah appreciate that, but thought we could deprecate project_task_dependency and project_timeline_task_dependency in favour of they now achieve the same in 15.0?

Comment (CarlosRoca13) [reply project_timeline/views/project_task_view.xml:20]: Sorry, I didn't know about the extra modules to add this. Added the feature :smile:

**Comment by pedrobaeza** [comment]
I'm having the same problem as stated in v14 about disappearing elements when moving them to unassigned, plus if I do double click in an item for editing it:

Peek 04-08-2022 09-43

**Comment by jguenat** [comment]
I don't have this traceback locally, seems like an issue with runboat. I tried to reset the runboat build.

**Comment by jguenat** [comment]
Hello @chrisandrewmann 
Do you also have the issues mentionned by @pedrobaeza ?
On my side it works as expected.

**Comment by chrisandrewmann** [comment]
@jguenat I don't see them myself either, I tested on a local instance and have also deployed to my production system.

**Comment by pedrobaeza** [comment]
Sorry, the concerns were already fixed:

/ocabot merge nobump

**Pull request merged**